executor.c

```diff
@@ -0,0 +1,90 @@
+//
+//  executor.c
+//  connectivi_graph
+//
+//  Created by Artem on 26/08/2022.
+//  Copyright © 2022 Artem. All rights reserved.
+//
+
+#include "finding_connectivity_components.h"
+
+
+Pair_int_int* read_grath_from_txt(char* file_path_name, int* points_cnt, int* arc_cnt){
+    Pair_int_int *v;
+    
+    FILE *in_file;
+    in_file = fopen(file_path_name, "r");
+    fscanf(in_file, "points count = %d", points_cnt);
+    fscanf(in_file, "\narcs count = %d", arc_cnt);
+    v = (Pair_int_int *)(calloc(*arc_cnt, sizeof(Pair_int_int)));
+    // graph is represented as a list of edges, which is necessary for Prim algorithm
+    // and significantly reduces the amount of stored data
+    for(int i = 0; i < *arc_cnt; ++i)
+        fscanf(in_file, "%d %d", &v[i].first, &v[i].second);
+    fclose(in_file);
+    return v;
+}
+
+
+int find_root(Tree_point *u_mass, Tree_point u){
+    while (u.me != u.my_root)
+        u = u_mass[u.my_root];
+    return u.me;
+}
+
+void create_point(Tree_point *u, int i){
+    u->me = i;
+    u->my_root = i;
+    u->cnt = 1;
+}
+
+void union_sets(Tree_point *u_mass, Pair_int_int v){
+    int r1, r2;
+    r1 = find_root(u_mass, u_mass[v.first]);
+    r2 = find_root(u_mass, u_mass[v.second]);
+    if(r1 == r2)
+        return;
+    Tree_point u1 = u_mass[r1];
```

Comment:
There is no need to create objects copy to modify it. References (pointers in C) may be used to simplify code instead of full copying.

Reply:
Agree, my current solution is very strange